desc/magnetic_fields.py

```diff
@@ -1846,6 +1853,328 @@ def from_surface(
         )
 
 
+class OmnigenousField(Optimizable, IOAble):
+    """A magnetic field with perfect omnigenity (but is not necessarily analytic).
+
+    Notes
+    -----
+    Doesn't conform to MagneticField API, as it only knows about :math:`|B|` in
+    computational coordinates, not vector B in lab coordinates.
+
+    Parameters
+    ----------
+    L_B : int
+        Resolution of the radial Chebyshev polynomials for magnetic well parameters.
+    M_B : int
+        Number of monotonic spline knots per surface of the magnetic well parameters.
+    L_x : int
+        Resolution of the radial Chebyshev polynomials for the omnigenity parameters.
+    M_x : int
+        Resolution of the Fourier series in eta for the omnigenity parameters.
+    N_x : int
+        Resolution of the Fourier series in alpha for the omnigenity parameters.
+    NFP : int
+        Number of field periods.
+    helicity : tuple, optional
+        Type of pseudo-symmetry (M, N). Default = toroidal contours (1, 0).
+    B_lm : ndarray, optional
+        Magnetic well parameters describing ||B||(ρ,η). These values are a flattened 2D
```

Comment:
flattened in what convention? i.e. you can `flatten("F")` or `flatten("C")` right? 

can this instead be made a 2D array and the flattening just happens internally?

Reply:
Added documentation to explain it is flattened in the default "C"-style order. 

The users should rarely be setting this themselves so I don't want to let them pass in a 2D array. It's the same thing as any other 2D basis set we use. For example, the coefficients of a `DoubleFourierSeries` are also a flattened 2D array where the rows/cols correspond to poloidal/toroidal modes. But we still make the user pass the boundary coefficients to a `FourierRZToroidalSurface` as a 1D array.